clients/gitlabrepo/checkruns.go

```diff
@@ -32,11 +33,20 @@ func (handler *checkrunsHandler) init(repourl *repoURL) {
 }
 
 func (handler *checkrunsHandler) listCheckRunsForRef(ref string) ([]clients.CheckRun, error) {
+	commit := regexp.MustCompile("^[a-f0-9]{40}$")
```

Comment:
let's only compile the regex once as this gets called multiple times, similar to like we do here:
https://github.com/ossf/scorecard/blob/663e1a9bad0f62b635cce71af8a1b3a0dabe7ba1/checks/raw/shell_download_validate.go#L60

On a similar note: let's add `A-Z` as a possibility too.